### PR TITLE
Finished start of Card and Hand Classes; Initalized cards and case fi…

### PR DESCRIPTION
**Pull request by gracekumagai**
Card Class is equivalent to Variable Class from CSP project, however, has a type <Room, Suspect, Weapon> and a name <Miss Scarlett, Lead Pipe, etc>. The name acts as the value. The type will/can be used to narrow down the domain of the name/value.

I have an idea on how to better prep the variables for the CSP, reach out before you start.

### CODE REVIEW

File: game.py [hunk 1]
```diff
@@ -38,23 +105,50 @@ def _made_accusation(self, player, suggestion):
...
-		init it with the things
+		Initialize a card with a type = <Room, Suspect, Weapon>
+		and a name = <Miss Scarlett, Lead Pipe, Lounge, ...>
 		'''
-		pass
+		self.type = type
+		self.name = name
+		self.player = None
+
+		self.dom = list(domain)
```
Comment (paulsuk) [inline game.py]: When do you use domain, as well as assignedType? Not sure about the implementation of these here

Comment (gracekumagai) [inline game.py]: tl dr: assignedType used to narrow down constraints, dom used to prune.

assignedType would be used to help specify the constraints if the type is known. For example, if the card must be <Room>. 

dom represents the permanent domain of the card. At the start of the game, the domain of someone elses card is the entire set of cards minus your own, and the domain of one of your cards is ONLY that cards true value.

Comment (paulsuk) [inline game.py]: then you don't need assignmentType since you already have a type variable

Comment (paulsuk) [inline game.py]: Also you never initialized domain of cards when you inited them in game

File: game.py [hunk 2]
```diff
@@ -1,18 +1,85 @@
...
+		self._make_case_file()
+
+
+	def _init_cards(self):
+		'''
+		Initialize all 21 cards
+		9 rooms, 6 weapons, 6 suspects
+		Return lists of each type
+		'''
+		#Init Rooms
```
Comment (paulsuk) [inline game.py]: You should also init their domain here

Comment (paulsuk) [inline game.py]: You could probably use some loop or iterable to do this @gracekumagai

Comment (gracekumagai) [inline game.py]: Done

File: game.py (hunk 1 above)
Comment (gracekumagai) [inline game.py]: Done

**Comment by gracekumagai** [comment]
CSP Model Idea: 3 csp's per player, representing the case file and 2 ghost hands of the other players.

File: game.py [hunk 3]
```diff
@@ -100,9 +208,11 @@ def make_move(self):
...
+
+		Update CSPs
 		'''
 		pass
 
 
-
-
+if __name__ == '__main__':
+	game = Game()
```
Comment (paulsuk) [inline game.py]: super small thing but add a blank line at the end of the file!

File: game.py [hunk 4]
```diff
@@ -1,18 +1,76 @@
...
+		'''
+		Initialize all 21 cards
+		9 rooms, 6 weapons, 6 suspects
+		Return lists of each type
+		'''
+		#Init Rooms
+		self.rooms = []
+		rooms = ['Conservatory', 'Hall', 'Lounge', 'Dining Room', 'Kitchen', 'Ballroom', 'Billiard Room', 'Library', 'Study']
+		for room in rooms:
+			r = Card('Room', room, [room])
```
Comment (paulsuk) [inline game.py]: When we are initing the cards, (even if they have a value) shouldn't their dom be all of the possible options, and have their curr_dom set as the assigned card?

@gracekumagai

Comment (gracekumagai) [inline game.py]: check out functions just pushed for curDom, its okay the way it is

**Pull request merged**